src/Quasar/FS/Mount.purs

```diff
@@ -19,35 +19,129 @@ module Quasar.FS.Mount where
 import Prelude
 
 import Data.Argonaut (Json, decodeJson, (.?))
+import Data.Const (Const(..))
 import Data.Either (Either(..))
-import Data.Generic.Rep (class Generic)
-import Data.Generic.Rep.Show (genericShow)
+import Data.Eq (class Eq1, eq1)
+import Data.Identity (Identity(..))
 import Data.Maybe (Maybe)
+import Data.Newtype (unwrap)
 import Data.Path.Pathy (DirName, FileName, dir, file, pathName, (</>))
-import Quasar.Mount.Type as MT
-import Quasar.Types (AnyPath, DirPath)
+import Quasar.Types (AnyPath, DirPath, FilePath)
 
-data Mount = Mount MT.MountType AnyPath
+data MountF f
+  = View (f FilePath)
+  | Module (f DirPath)
+  | MongoDB (f DirPath)
+  | Couchbase (f DirPath)
+  | MarkLogic (f DirPath)
+  | SparkHDFS (f DirPath)
+  | SparkLocal (f DirPath)
+  | Mimir (f DirPath)
+  | Unknown String (f AnyPath)
 
-derive instance eqMount ∷ Eq Mount
-derive instance ordMount ∷ Ord Mount
-derive instance genericMount ∷ Generic Mount _
-instance showMount ∷ Show Mount where show = genericShow
+type Mount = MountF Identity
+type MountType = MountF (Const Unit)
+
+
+instance eqMount ∷ (Eq1 f) => Eq (MountF f) where
+  eq inX inY =  case inX of
+    View x
+      | View y <- inY -> eq1 x y
+      | otherwise -> false
+    Module x
+      | Module y <- inY -> eq1 x y
+      | otherwise -> false
+    MongoDB x
+      | MongoDB y <- inY -> eq1 x y
+      | otherwise -> false
+    Couchbase x
+      | Couchbase y <- inY -> eq1 x y
+      | otherwise -> false
+    MarkLogic x
+      | MarkLogic y <- inY -> eq1 x y
+      | otherwise -> false
+    SparkHDFS x
+      | SparkHDFS y <- inY -> eq1 x y
+      | otherwise -> false
+    SparkLocal x
+      | SparkLocal y <- inY -> eq1 x y
+      | otherwise -> false
+    Mimir x
+      | Mimir y <- inY -> eq1 x y
+      | otherwise -> false
+    Unknown xName x
+      | Unknown yName y <- inY -> eq1 x y && eq xName yName
+      | otherwise -> false
```

Comment:
Why use pattern guards like this rather than:
``` purescript
eq inX inY = case inX, inY of
  View x, View y -> eq1 x y
  View x, _ -> false
  Module x, Module y -> eq1 x y
  Module x, _ -> false
```
?

Reply:
You just need to specify name of variant 3 times vs 2
``` purs
View x, View y -> eq1 x y
View x, _ -> false
```
vs
``` purs
View x
  | View y <- inY -> eq1 x y
  | otherwise -> false
```

Do you see other reasons for not using what I have?

Reply:
Yes, it's weird frankly. :laughing:

Reply:
Output is not that good for it too :/
like I could not understand what's happening there :D
https://gist.github.com/safareli/f1446b4daee3b546c1fdcd2efe42223b

Reply:
I'm feeling i should open an issue as it looks way to complicated.

Reply:
I prefer 

```purescript
View x, View y -> eq1 x y
View x, _ -> false
```

It's clear and simple what's going on, and it's only two lines with less syntax, imo.

Reply:
It's because each pattern guard desugars into another case, pretty much. It's like writing nested cases.

Reply:
Ah I guess it's because the `|` is guard thing and it should happen only if guard before it failed so they need to be wrapped in function?

Reply:
A mistake in my proposal that has been repeated everywhere since:
``` purescript
View x, View y -> eq1 x y
View x, _ -> false
```
Can be:
``` purescript
View x, View y -> eq1 x y
View _, _ -> false
```

Reply:
you can write it  in two lines too
```
View x | View y <- inY -> eq1 x y
  | otherwise -> false
```
or even one:
```
View x | View y <- inY -> eq1 x y | otherwise -> false
```

But I it's output is not what we want to have